cmake/pcl_find_avx.cmake

```diff
@@ -0,0 +1,37 @@
+###############################################################################
+# Check for the presence of AVX and figure out the flags to use for it.
+macro(PCL_CHECK_FOR_AVX)
+  set(AVX_FLAGS)
+
+  include(CheckCXXSourceRuns)
+  
+  check_cxx_source_runs("    
+    #include <immintrin.h>
+    int main()
+    {
+      __m256i a = {0};
+      a = _mm256_abs_epi16(a);
+      return 0;
+    }"
+  HAVE_AVX2)
+
+  if(NOT HAVE_AVX2)
```

Comment:
Should `HAVE_AVX` escape outside of this macro? If yes add `set(HAVE_AVX ${HAVE_AVX2})` before. Otherwise change the macro to an function (don't forget than to include `PARENT_SCOPE` `set(AVX_FLAGS "/arch:AVX2 PARENT_SCOPE)`)

Reply:
HAVE_AVX(2) are only supposed to used in the macro. The only part that should "exit" the macro is AVX_FLAGS, which are used in the main CMakeLists.

Reply:
In this case `function` would be better than `macro` to prevent someone is using this variable later.